applications/crossbar/src/modules/provisioner_util.erl

```diff
@@ -12,64 +12,39 @@
 
 -include("../crossbar.hrl").
 
--export([get_mac_address/1]).
--export([get_old_mac_address/1]).
 -export([maybe_provision/1]).
 -export([maybe_delete_provision/1]).
 -export([maybe_update_account/1]).
 -export([maybe_delete_account/1]).
 -export([maybe_send_contact_list/1]).
 -export([get_provision_defaults/1]).
--export([delete_full_provision/2]).
 -export([is_mac_address_in_use/2]).
+-export([maybe_sync_sip_data/2]).
 
 -define(MOD_CONFIG_CAT, <<(?CONFIG_CAT)/binary, ".devices">>).
 -define(PROVISIONER_CONFIG, <<"provisioner">>).
 -define(TEMPLATE_ATTCH, <<"template">>).
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
--spec get_mac_address(cb_context:context()) -> 'undefined' | string().
+
+-spec get_mac_address(cb_context:context()) -> api_binary().
 get_mac_address(Context) ->
-    case wh_json:get_ne_value(<<"mac_address">>, cb_context:doc(Context)) of
-        'undefined' -> 'undefined';
-        MACAddress ->
-            re:replace(wh_util:to_list(MACAddress)
-                       ,"[^0-9a-fA-F]"
-                       ,""
-                       ,[{'return', 'list'}
-                         ,'global'
-                        ]
-                      )
-    end.
+    MACAddress = wh_json:get_ne_binary_value(<<"mac_address">>, cb_context:doc(Context)),
+    cleanse_mac_address(MACAddress).
 
-%%--------------------------------------------------------------------
-%% @public
-%% @doc
-%%
-%% @end
-%%--------------------------------------------------------------------
--spec get_old_mac_address(cb_context:context()) -> 'undefined' | string().
+-spec get_old_mac_address(cb_context:context()) -> api_binary().
 get_old_mac_address(Context) ->
-    case cb_context:fetch(Context, 'db_doc') of
-        'undefined' -> 'undefined';
-        JObj ->
-            case wh_json:get_ne_value(<<"mac_address">>, JObj) of
-                'undefined' -> 'undefined';
-                MACAddress ->
-                    re:replace(wh_util:to_list(MACAddress)
-                               ,"[^0-9a-fA-F]"
-                               ,""
-                               ,[{'return', 'list'}
-                                 ,'global'
-                                ]
-                              )
-            end
-    end.
+    MACAddress =
+        case cb_context:fetch(Context, 'db_doc') of
+            'undefined' -> 'undefined';
+            JObj ->
+                wh_json:get_ne_value(<<"mac_address">>, JObj)
+        end,
+    cleanse_mac_address(MACAddress).
+
+-spec cleanse_mac_address(api_binary()) -> api_binary().
+cleanse_mac_address('undefined') -> 'undefined';
+cleanse_mac_address(MACAddress) ->
+    binary:replace(MACAddress, <<":">>, <<"">>, ['global']).
```

Comment:
This doesn't seem to be the equivalent to the former regex. Stripping non-alphanumeric characters is different from only removing `:`.